scripts/src/java/org/oppia/android/scripts/build/FilterPerLanguageResources.kt

```diff
@@ -0,0 +1,148 @@
+package org.oppia.android.scripts.build
+
+import com.android.aapt.Resources.ConfigValue
+import com.android.aapt.Resources.Entry
+import com.android.aapt.Resources.Package
+import com.android.aapt.Resources.ResourceTable
+import com.android.aapt.Resources.Type
+import org.oppia.android.app.model.LanguageSupportDefinition
+import org.oppia.android.app.model.SupportedLanguages
+import java.io.File
+import java.util.Locale
+import java.util.zip.ZipEntry
+import java.util.zip.ZipFile
+import java.util.zip.ZipOutputStream
+
+/**
+ * Script for filtering unsupported languages out of Android app bundle resource tables.
+ *
+ * Usage:
+ *   bazel run //scripts:filter_per_language_resources -- \\
+ *     <path_to_input_module> <path_to_output_module>
+ *
+ * The input module is expected to be a well-formed AAB module zip archive, and include a
+ * support_languages.pb asset to provide guidance on which languages should be kept.
+ *
+ * Arguments:
+ * - path_to_input_module: path to the input AAB module zip to be processed.
+ * - path_to_output_module: path to the output AAB module zip that will be written.
+ *
+ * Example:
+ *   bazel run //scripts:filter_per_language_resources -- \\
+ *     $(pwd)/bazel-bin/oppia_dev_raw_module.zip $(pwd)/oppia_dev_updated_module.zip
+ */
+fun main(vararg args: String) {
```

Comment:
Would it be better to pass the args separately and make sure there are exactly 2 args rather than >=2? The use of args[0], args[1] seems a bit less clear, too; would prefer them being named.

Reply:
The exact equality checking seems reasonable, and I added names when passing these to filterPerLanguageResources to help provide the potentially missing context. However, I can't change main() since it always takes an array of args (the vararg is just a syntactical variation of taking an array; both are mostly equivalent).

Reply:
Now that the changes are pushed, marking this as resolved since you've approved the PR.